servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletPrintWriter.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.servlet;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.util.Formatter;
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+/**
+ *  Printing flow.
+ */
+final class ServletPrintWriter extends PrintWriter {
+    private final String lineSeparator = System.lineSeparator();
+    private final DefaultServletHttpResponse response;
+
+    private boolean error;
+
+    ServletPrintWriter(DefaultServletHttpResponse response, OutputStream out) {
+        super(out);
+        requireNonNull(response, "response");
+        this.response = response;
+    }
+
+    @Override
+    public void flush() {
+    }
+
+    @Override
+    public void close() {
+        response.close();
+    }
+
+    @Override
+    public boolean checkError() {
+        return error;
+    }
+
+    @Override
+    protected void setError() {
+        error = true;
+    }
+
+    @Override
+    protected void clearError() {
+        error = false;
+    }
+
+    @Override
+    public void write(int c) {
+        write(String.valueOf(c));
+    }
+
+    @Override
+    public void write(char[] buf, int off, int len) {
```

Comment:
Is there any reason that you overridden methods in this class?
It seems like we can just use the methods in `PrintWriter`?

Reply:
Method `PrintWriter.write()` may be call to `OutputStream.write()`. I will check all write methods If there is any method that doesn't need to override, i will remove code.